Make log optional

not-fl3 opened the pull request:
By default, with the "log" feature on "zune-core" will re-export the log crate.
Without "log", "zune-core" exports some stubs over `eprintln!`. 
The only change required for "zune-*" crate: use `zune_core::log::trace` instead of `log::trace`.

To remove "log-rs" dependency in zune-image user's Cargo.toml:
```
zune-jpeg = {version = "0.3", default-features = false, features = ["x86", "neon", "std"]}
```
I do not really like that it requires "x86" etc, but I guess its fine? Not having `log` as a default featulre will be a breaking change - for everyone who actually used `log-rs`, all the logs from `zune` will suddenly disappear, not ideal.

Let me know if the idea is OK, I'll finish this PR with the rest of "zune-*" crates.

etemesi254 commented:
yea, it is ideal. I'm planning on a breaking change anyways since i switched from requiring slices for input to a generic trait T

not-fl3 commented:
> yea, it is ideal. I'm planning on a breaking change anyways since i switched from requiring slices for input to a generic trait T

perfect! pushed the rest of the crates with the "log" being off by default. 

The only thing that concerned me - I changed log for zune_core in zune-imageprocs. Do not really know whats the relashionship between zune-imageprocs and zune_core and if that was a right thing to do.

etemesi254 commented:
> The only thing that concerned me - I changed log for zune_core in zune-imageprocs. Do not really know whats the relashionship between zune-imageprocs and zune_core and if that was a right thing to do.

The reason it never used `zune-core` was that there was no need for them to interact.

The imageprocs does image processing stuff- blurs, color manipulation etc. core mainly provides decoding and encoding routines like byte readers and writers.

etemesi254 commented on zune-core/src/log.rs:
I'm not particularly okay with printing to standard output. I.e it's a library and it wouldn't make sense to print warnings for something we can't really influence where it would be used.

I think if the user doesn't want logging, let us just silently fail

not-fl3 replied:
done! warn and error are now panic and anything else is just nothing. 

tbh as a user I would not mind info! and warn! to go into stderr
